src/odemis/acq/stream/_base.py

```diff
@@ -1302,7 +1302,8 @@ def getPixelCoordinates(self, p_pos):
         tform = AffineTransform(matrix, translation)
         pixel_pos_c = tform.inverse().apply(p_pos)
         # a "-" is used for the y coordinate because Y axis has the opposite direction in physical coordinates
-        pixel_pos = int(pixel_pos_c[0] + size[0] / 2), - int(pixel_pos_c[1] - size[1] / 2)
+        pixel_pos = (int(math.floor(pixel_pos_c[0] + size[0] / 2)),
+                     int(math.floor(- (pixel_pos_c[1] - size[1] / 2))))
```

Comment:
Is `floor` not problematic for the positive scenario, where going outside of the image in the positive axes now makes it "in image"?

Reply:
```suggestion
                     # Calculate pixel positions
                     x_pos_c = pixel_pos_c[0] + size[0] / 2
                     y_pos_c = -(pixel_pos_c[1] - size[1] / 2)

                    # Apply floor or ceil based on the sign
                    x_pos = math.floor(x_pos_c) if x_pos_c < 0 else math.ceil(x_pos_c)
                    y_pos = math.floor(y_pos_c) if y_pos_c < 0 else math.ceil(y_pos_c)
                    pixel_pos = (x_pos, y_pos)
```

Maybe

Reply:
So this would functionally alter the in range values (since `int` cast 'rounds' towards zero, but now the values are ceiled). To make if functionally equivalent, the default (only for this range, since doesn't hold for negative ranges) should be `floor`. I can also imagine that we want to `math.round` instead, before casting to `int`.

Reply:
No. We just want to use `floor()` all the time. That's it.
To give an example, using a little extreme, but obvious values, imagine an image of resolution 1x1. In the "pixel" coordinates, the image goes from 0.0 -> 1.0 . And any position between 0.0 and 1.0 is on pixel 0. Position -0.001 is outside the image. Position 1.001 is outside the image. That's why `floor()` does exactly what we need.

Reply:
I see now! Might also be helpful to add something similar as a comment, since this explanation was clarifying.